enricher/standard/src/main/java/io/fabric8/maven/enricher/standard/ConfigMapEnricher.java

```diff
@@ -91,9 +92,12 @@ private void addConfigMapFromXmlConfigurations(KubernetesListBuilder builder) {
         final Map<String, String> configMapFromConfiguration;
         try {
             configMapFromConfiguration = createConfigMapFromConfiguration(configMap);
-            if(!configMapFromConfiguration.isEmpty() && !checkIfItemExists(builder, "xmlconfig")) {
+            String configMapName = (configMap == null || configMap.getName() == null || configMap.getName().trim().isEmpty()) ? "xmlconfig" : configMap.getName().trim();
+            log.debug("configMapName :: ".concat(configMapName));
+
+            if(!configMapFromConfiguration.isEmpty() && !checkIfItemExists(builder, configMapName)) {
```

Comment:
I think the reason why config map is not getting created is due to `checkIfItemExists(builder, configMapName)` returning true always(ConfigMap name was hardcoded earlier, that's why it returned false, and at this point already created Deployment & Service are also created with same name). I think we should remove this method or maybe change it check for kind also

Reply:
Or maybe change checkIfItemExists to this:
```
    private boolean checkIfItemExists(KubernetesListBuilder builder, String name) {
        return builder.buildItems().stream().anyMatch(item -> item.getKind().equals("CongifMap") && item.getMetadata().getName().equals(name));
    }
```